Adding a 'not' filter

scotwk opened the pull request:
@kapilt - is this what you're going for, or is there more to this filter than this?

I've tested it and it works in a basic test, but my unit test doesn't work because the tests don't utilize a query manager, so I have to figure out how to write a `__call__` method (or modify the tests to use a query manager).

kapilt commented on c7n/filters/core.py:
```diff
@@ -200,6 +203,27 @@ def process(self, resources, events=None):
+        self.filters = registry.parse(self.data.values()[0], manager)
+        self.manager = manager
+
+    def process(self, resources, events=None):
```
event is singular at the moment

scotwk replied:
@kapilt - I just copied from the And filter.  Should that one be singular also?

kapilt commented:
yup, this was the goal, thanks!

scotwk commented:
@kapilt - I got all the tests passing, but I'm trying to figure out if I need to special case `not` in filters/vpc.py and filters/related.py.  Also, I should probably add a test that hits the `process_set function in the new `Not` filter for coverage purposes.

scotwk commented:
@kapilt - I added a test to complete coverage of the new code.  Looks like the `and` and `or` in filters/vpc.py and filters/related.py are independent of what I added.

Should be good to go now.
